gms/versioned_value.hh

```diff
@@ -95,7 +95,7 @@ public:
     }
 
     static sstring version_string(const std::initializer_list<sstring>& args) {
-        return ::join(sstring(versioned_value::DELIMITER_STR), args);
+        return fmt::to_string(fmt::join(args, std::string_view(versioned_value::DELIMITER_STR)));
```

Comment:
Is the cast to string_view really necessary here?

Reply:
yes. `versioned_value::DELIMITER_STR` is defined as an array of char, i.e., 
```c++
constexpr const char versioned_value::DELIMITER_STR[];
```
which is not a string-like thing. we can improve this part later. but i wanted to focus on the `fmt::join()` related changes in this PR.